### PR TITLE
Add client with TLS transport

### PR DESCRIPTION
**Pull request by agalitsyn**
Add TLS settings to:
 - [x] helm repo add
 - [x] helm repo update
 - [x] helm fetch

### CODE REVIEW

File: pkg/util/crypto.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+}
+
+// CertFromFilePair returns an tls.Certificate containing the
+// certificates public/private key pair from a pair of given PEM-encoded files.
+// Returns an error if the file could not be read, a certificate could not
+// be parsed, or if the file does not contain any certificates
+func CertFromFilePair(certFile, keyFile string) (tls.Certificate, error) {
+	cert, err := tls.LoadX509KeyPair(certFile, keyFile)
+	if err != nil {
+		return tls.Certificate{}, fmt.Errorf("can't load key pair from cert %s and key %s", certFile, keyFile)
```
Comment (klizhentas) [inline pkg/util/crypto.go]: you would not need to create empty object if you changed the signature of the function to return `*tls.Certificate` instead

**Review by klizhentas — APPROVED** [review]

File: pkg/util/crypto.go (hunk 1 above)
Comment (agalitsyn) [reply pkg/util/crypto.go]: true, thanks

**Comment by technosophos** [comment]
I am so happy to see you guys working on this. I am hoping we could upstream this sort of work back into Helm because it's something we really want and need.

I know @adamreese is already reviewing the details,, but he's out for Thanksgiving. I can give a few brief things that I noticed:

1. I do think this is the right general direction, and am enthusiastic about finally getting good SSL?TLS support into Helm.
2. For the SSL/TLS params, it would be nice to have the option to use env vars.
3. Adam has been working on a pattern for shared flags, but I can't remember if it's finished. As you saw in the code, it's getting to be a problem for us. We hate copying the same flag set from one command to another.
4. As dumb as it may sound, we have a rule against naming a package `util`, though `httputil` or `tlsutil` is just fine.

I'll let Adam weigh in on the mechanics of the HTTP client. IIRC, he designed that part.

**Comment by klizhentas** [comment]
@agalitsyn check out the comments form @technosophos above

**Pull request merged**